app/forms/sections/items/ItemDesignationOfOriginFormProvider.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.sections.items
+
+import forms.XSS_REGEX
+import forms.mappings.Mappings
+import forms.sections.items.ItemDesignationOfOriginFormProvider._
+import models.sections.items.ItemGeographicalIndicationType.{ProtectedDesignationOfOrigin, ProtectedGeographicalIndication}
+import models.sections.items.{ItemDesignationOfOriginModel, ItemGeographicalIndicationType}
+import play.api.data.Form
+import play.api.data.Forms.{mapping, text => playText}
+import uk.gov.voa.play.form.Condition
+import uk.gov.voa.play.form.ConditionalMappings.mandatoryIf
+
+import javax.inject.Inject
+
+class ItemDesignationOfOriginFormProvider @Inject() extends Mappings {
+
+  //scalastyle:off
+  def apply(epc: String): Form[ItemDesignationOfOriginModel] =
+    Form(
+      mapping(
+        geographicalIndicationField -> enumerable[ItemGeographicalIndicationType](geographicalIndicationChoiceRequiredError),
+
+        protectedDesignationOfOriginTextField -> mandatoryIfOptionSelectedAndInputNonEmpty(
+          geographicalIndicationField,
+          ProtectedDesignationOfOrigin.toString,
+          protectedDesignationOfOriginTextField,
+          playText()
+            .verifying(maxLength(geographicalIndicationIdentificationMaxLength, geographicalIndicationIdentificationLengthError))
+            .verifying(regexp(XSS_REGEX, geographicalIndicationIdentificationInvalidError))
+            .transform[String](
+              _.replace("\n", " ")
+                .replace("\r", " ")
+                .replaceAll(" +", " ")
+                .trim,
+              identity
+            )
+        ),
+
+        protectedGeographicalIndicationTextField -> mandatoryIfOptionSelectedAndInputNonEmpty(
+          geographicalIndicationField,
+          ProtectedGeographicalIndication.toString,
+          protectedGeographicalIndicationTextField,
+          playText()
+            .verifying(maxLength(geographicalIndicationIdentificationMaxLength, geographicalIndicationIdentificationLengthError))
+            .verifying(regexp(XSS_REGEX, geographicalIndicationIdentificationInvalidError))
+            .transform[String](
+              _.replace("\n", " ")
+                .replace("\r", " ")
+                .replaceAll(" +", " ")
+                .trim,
```

Comment:
We have this in a lot of places; I do wonder whether it's worth creating a partial function in a `trait` that can get mixed into some of these forms. So that you could just call a `trimTextArea` sort of function.

Reply:
normalisedSpaceText does exist but it uses the custom `text` mapping, I'll split out the replace logic so it can be called separately for those components that don't conform to the usual pattern